From c7d7c2ba242a8624c63a0bac7f53672bdf3ecf1c Mon Sep 17 00:00:00 2001
From: Emmanuel Berthier <emmanuel.berthier@intel.com>
Date: Wed, 16 Sep 2015 15:08:34 +0200
Subject: [PATCH] Correct interruption unmasking issue

CLI/STI are not needed and generate unexpected Interrupt unmasking.
Replace this durty asm code by the right clean kernel function.

Change-Id: Icbfa2d7b3d08f3a39b9d2de569115cd959c2948f
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-2561
Signed-off-by: Emmanuel Berthier <emmanuel.berthier@intel.com>
---
 arch/x86/kernel/smp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kernel/smp.c b/arch/x86/kernel/smp.c
index bfec8ab..c86332a 100644
--- a/arch/x86/kernel/smp.c
+++ b/arch/x86/kernel/smp.c
@@ -123,7 +123,7 @@ static inline void store_regs(struct pt_regs *regs)
 	crash_setup_regs(&get_cpu_var(cpu_regs), regs);
 
 	/* Flush CPU cache */
-	asm("cli;wbinvd;sti");
+	wbinvd();
 }
 
 /*
-- 
1.9.1

